ipa-core/src/protocol/ipa_prf/validation_protocol/proof_generation.rs

```diff
@@ -0,0 +1,461 @@
+            context::dzkp_field::BLOCK_SIZE,
+            ipa_prf::{
+                malicious_security::prover::{LargeProofGenerator, SmallProofGenerator},
+                validation_protocol::proof_generation::BatchToVerify,
+            },
+        },
+        secret_sharing::{replicated::ReplicatedSecretSharing, SharedValue},
+        test_executor::run,
+        test_fixture::{Runner, TestWorld},
+    };
+
+    impl Default for Fp61BitPrime {
+        fn default() -> Self {
+            Fp61BitPrime::ZERO
+        }
+    }
+
+    #[test]
+    fn generate_verifier_batch() {
+        run(|| async move {
+            let world = TestWorld::default();
+
+            let mut rng = thread_rng();
+
+            // each helper samples a random value h
+            // which is later used to generate distinct values across helpers
+            let h = Fp61BitPrime::truncate_from(rng.gen_range(0u128..100));
+
+            let result = world
+                .semi_honest(h, |ctx, h| async move {
+                    let h = Fp61BitPrime::truncate_from(h.left().as_u128() % 100);
+                    // generate blocks of UV values
+                    // generate u values as (1h,2h,3h,....,10h*BlockSize) split into Blocksize chunks
+                    // where BlockSize = 32
+                    // v values are identical to u
+                    let uv_tuple_vec = (0usize..100)
+                        .map(|i| {
+                            (
+                                (BLOCK_SIZE * i..BLOCK_SIZE * (i + 1))
+                                    .map(|j| {
+                                        Fp61BitPrime::truncate_from(u128::try_from(j).unwrap()) * h
+                                    })
+                                    .collect::<[Fp61BitPrime; BLOCK_SIZE]>(),
+                                (BLOCK_SIZE * i..BLOCK_SIZE * (i + 1))
+                                    .map(|j| {
+                                        Fp61BitPrime::truncate_from(u128::try_from(j).unwrap()) * h
+                                    })
+                                    .collect::<[Fp61BitPrime; BLOCK_SIZE]>(),
+                            )
+                        })
+                        .collect::<Vec<_>>();
+
+                    // generate and output VerifierBatch together with h value
+                    (
+                        h,
+                        BatchToVerify::generate_batch_to_verify(ctx, uv_tuple_vec.into_iter())
+                            .await,
+                    )
+                })
+                .await;
+
+            // proof from first party
+            simple_proof_check(result[0].0, &result[2].1, &result[1].1);
+
+            // proof from second party
+            simple_proof_check(result[1].0, &result[0].1, &result[2].1);
+
+            // proof from third party
+            simple_proof_check(result[2].0, &result[1].1, &result[0].1);
+        });
+    }
+
+    fn simple_proof_check(
```

Comment:
I don't follow this. This is not how we will eventually verify these proofs, it's not in the paper, and it's not clear to me how this works. I would rather not test the code in this way.

Rather than add all of this extra logic that does a novel approach - which requires reading and understanding, perhaps we should just defer testing this until we have more of the verifier code in place?

Reply:
Yes I think that makes a lot of sense. Maybe we keep this test for now but once we have the verifier in place, we can delete it?

Reply:
If we plan to just delete it anyway, I would rather not add it in the first place. Let's keep the diff history simpler.